test/test_linalg.py

```diff
@@ -232,32 +225,35 @@ def check_singular_values(a, sv):
         # that is why we use `cond=1.0`, the mean to cut roughly half of all
         # the singular values and compare whether torch.linalg.lstsq agrees with
         # SciPy and NumPy.
-        cond = (None, 1.0)
+        # if rcond is True then set value for it based on the used algorithm
+        # rcond == -1 or any other negative value forces LAPACK to use machine precision tolerance
+        rconds = (None, True, -1)
+
+        for batch, matrix_size, driver, rcond in itertools.product(batches, matrix_sizes, drivers, rconds):
+            if rcond and driver != 'gels':
```

Comment:
These two if statements seem like they could be written in a more straightforward way? Maybe I'm reading this incorrectly

Reply:
Looking at it again I see that is indeed confusing and not correct. I refined the if statements:
```python
# keep the rcond value if it is None or -1, set the driver specific value if it is True
if rcond and rcond != -1:
    if driver in ('gelss', 'gelsd'):
        # SVD based algorithm; set to zero roughly half of all the singular values
        rcond = 1.0
    else:
        # driver == 'gelsy'
        # QR based algorithm; setting the value too high might lead to non-unique solutions and flaky tests
        rcond = 1e-3

# specifying rcond value has no effect for gels driver so no need to run the tests again
if driver == 'gels' and rcond != None:
    continue
```